packages/dnb-eufemia/src/extensions/forms/Field/SelectCurrency/SelectCurrency.tsx

```diff
@@ -0,0 +1,377 @@
+    },
+    [getCurrencyObjectByIso, handleChange]
+  )
+
+  const fillData = useCallback(() => {
+    if (!wasFilled.current) {
+      wasFilled.current = true
+      dataRef.current = getCurrencyData({
+        lang: langRef.current,
+        filter,
+        sort: ccFilter as Extract<CurrencyFilterSet, 'Prioritized'>,
+      })
+      forceUpdate()
+    }
+  }, [ccFilter, filter, forceUpdate])
+
+  const onFocusHandler = useCallback(
+    ({ updateData }) => {
+      fillData()
+      updateData(dataRef.current)
+      handleFocus()
+    },
+    [fillData, handleFocus]
+  )
+
+  const onTypeHandler = useCallback(
+    ({ value: currentValue, setHidden, event }) => {
+      // Handle browser autofill/autocomplete
+      if (typeof event?.nativeEvent?.data === 'undefined') {
+        const search = currentValue.toLowerCase()
+        const currency = currencies.find(({ i18n }) =>
+          Object.values(i18n).some((s) => s.toLowerCase().includes(search))
+        )
+        if (currency?.iso) {
+          setHidden()
+          handleChange(currency.iso)
+        }
+      }
+    },
+    [handleChange]
+  )
+
+  useMemo(() => {
+    if (path || itemPath) {
+      setDisplayValue(
+        getCurrencyObjectByIso(value)?.i18n?.[langRef.current]
+      )
+    }
+  }, [getCurrencyObjectByIso, itemPath, path, setDisplayValue, value])
+
+  const fieldBlockProps: FieldBlockProps = {
+    forId: id,
+    className: classnames('dnb-forms-field-select-currency', className),
+    label,
+    width:
+      width === 'stretch' || fieldBlockContext?.composition
+        ? width
+        : undefined,
+    contentWidth: width !== false ? width : undefined,
+    ...pickSpacingProps(props),
+  }
+
+  return (
+    <FieldBlock {...fieldBlockProps}>
+      <Autocomplete
+        id={id}
+        placeholder={placeholder}
+        input_icon={false}
+        data={dataRef.current}
+        value={typeof value === 'string' ? value : null}
+        disabled={disabled}
+        size={size}
+        on_show={fillData}
+        on_focus={onFocusHandler}
+        on_blur={handleBlur}
+        on_change={handleCurrencyChange}
+        on_type={onTypeHandler}
+        stretch
+        status={hasError ? 'error' : undefined}
+        show_submit_button
+        keep_selection
+        // Todo: look into transaction-amount
+        autoComplete="transaction-currency"
+        no_animation={props.noAnimation}
+        {...htmlAttributes}
+      />
+    </FieldBlock>
+  )
+}
+
+type GetCurrencyData = {
+  lang?: CurrencyLang
+  filter?: Props['filterCurrencies']
+  sort?: Extract<CurrencyFilterSet, 'Prioritized'>
+  makeObject?: (
+    currency: CurrencyType,
+    lang: string
+  ) => {
+    selectedKey: string
+    selected_value?: string
+    content: string
+  }
+}
+
+export function getCurrencyData({
+  lang = 'nb',
+  filter = null,
+  sort = null,
+  makeObject = (currency: CurrencyType, lang: string) => {
+    const content = currency.i18n[lang] ?? currency.i18n.en
```

Comment:
```suggestion
    const content = [currency.i18n[lang] ?? currency.i18n.en, currency.iso]
```

I think we should add `iso` as well. It should be possible to search for e.g. "nok".

Reply:
I will add this. Just have to update a few tests and examples before pushing the commit.

Reply:
Done in commit https://github.com/dnbexperience/eufemia/pull/4770/commits/ee979850f923e07e7a656318860c8ed85e4c5dc8

Reply:
When dropdown is open, it looks like:

<img width="394" alt="image" src="https://github.com/user-attachments/assets/e56e5561-0027-4e63-a2c2-0642c043c338" />


When selected it looks like:

<img width="587" alt="image" src="https://github.com/user-attachments/assets/e5d4bf58-7bdc-4cce-aa9b-e357a19456d1" />